wrapanapi/systems/lenovo.py

```diff
@@ -112,6 +120,25 @@ def list_servers(self):
         self._servers_list = inventory
         return inventory
 
+    def list_switches(self):
+        raw_switches = self._service_instance(LenovoSwitch.API_PATH).get('switchList', [])
+        return [LenovoSwitch(data, system=self) for data in raw_switches]
+
+    def get_switch(self, uuid):
+        switch_data = self._service_instance("{}/{}".format(LenovoSwitch.API_PATH, uuid.lower()))
+        switch = None
+        if switch_data:
+            switch = LenovoSwitch(switch_data, system=self)
+        return switch
+
+    def delete_switch(self, uuid):
+        switch = self.get_switch(uuid)
```

Comment:
I like the pattern that you've established now in `get_switch` to handle no match for a uuid and return `None`, but that needs to be accounted for here, where we're assigning `switch` to a value that is potentially `None`.

The next statement, `switch.cleanup()`, will raise an AttributeError on not matching a uuid, with the message `NoneType has not attribute 'cleanup'`.

Do you want this error to continue to the caller? I think it would be better to raise a more meaningful exception when this occurs.